src/index.js

```diff
@@ -1,126 +1,32 @@
-import Tracker from './tracker/index.js';
-
-const args = process.argv.slice(2);
-const refilterOnly = args.includes('--refilter-only');
-const schedule = args.includes('--schedule');
-const extraArgs = args.filter(arg => !arg.startsWith('--'));
-
-const __dirname = path.dirname(fileURLToPath(import.meta.url));
-
-(async function startOpenTermsArchive() {
-  const { versionsStorageAdapter, snapshotsStorageAdapter } = initStorageAdapters();
-
-  const archivist = new Archivist({
-    storage: {
-      versions: versionsStorageAdapter,
-      snapshots: snapshotsStorageAdapter,
-    },
-  });
-
-  archivist.attach(logger);
-
-  await archivist.initialize();
-
-  logger.info('Start Open Terms Archive\n');
-
-  let serviceIds;
-
-  if (extraArgs.length) {
-    serviceIds = extraArgs.filter(serviceId => {
-      const isServiceDeclared = archivist.serviceDeclarations[serviceId];
-
-      if (!isServiceDeclared) {
-        logger.warn(`Parameter "${serviceId}" was interpreted as a service ID to update, but no matching declaration was found; it will be ignored`);
-      }
-
-      return isServiceDeclared;
-    });
-  }
-
-  await archivist.refilterAndRecord(serviceIds);
-
-  if (refilterOnly) {
-    return;
-  }
-
-  if (process.env.NODE_ENV === 'production') {
-    archivist.attach(new Notifier(archivist.serviceDeclarations));
-  }
 
-  if (process.env.GITHUB_TOKEN) {
-    archivist.attach(new Tracker());
-  }
-
-  await archivist.trackChanges(serviceIds);
-
-  if (!schedule) {
-    return;
-  }
-
-  logger.info('The scheduler is running…');
-
-  const TRACK_CHANGES_HOURS_INTERVAL = 24;
-  const RELEASE_HOURS_INTERVAL = 24 * 7;
-
-  logger.info(`Documents will be tracked every ${TRACK_CHANGES_HOURS_INTERVAL} hours\n`);
-  setInterval(async () => {
-    await archivist.trackChanges(serviceIds);
-  }, TRACK_CHANGES_HOURS_INTERVAL * 60 * 60 * 1000);
-
-  if (config.get('dataset.publish')) {
-    logger.info(`A release will be published every ${RELEASE_HOURS_INTERVAL} hours\n`);
-
-    setInterval(async () => {
-      logger.info('Start creating the release…');
-      await publishRelease();
-      logger.info('Release published');
-    }, RELEASE_HOURS_INTERVAL * 60 * 60 * 1000);
-  }
-}());
-
-function initStorageAdapters() {
-  let versionsStorageAdapter;
-
-  if (config.has('recorder.versions.storage.git')) {
-    versionsStorageAdapter = new GitAdapter({
-      ...config.get('recorder.versions.storage.git'),
-      path: path.resolve(__dirname, '../', config.get('recorder.versions.storage.git.path')),
-      fileExtension: 'md',
-    });
-  }
+export function instantiateVersionsStorageAdapter() {
+  return instantiateStorageAdapter('versions', 'md');
+}
 
-  if (config.has('recorder.versions.storage.mongo')) {
-    versionsStorageAdapter = new MongoAdapter(config.get('recorder.versions.storage.mongo'));
-  }
+export function instantiateSnapshotsStorageAdapter() {
+  return instantiateStorageAdapter('snapshots', 'html');
```

Comment:
It is overridden when it is a PDF file. But this should be refactored in another PR as it is a legacy.

Reply:
But since this function is exported, how can the override happen? I understand how the Core can, in its use of this adapter, override it when needed. But if I'm a third party using this adapter, how would it work when I try to load a snapshot that is a PDF file?

I understand we introduce this in this PR because we use the versions adapter in the dataset export script. If we don't need to consume the snapshots adapter from the outside of the app now, let's just not export this function and add a comment explaining that the html one is overridden by a pdf one in the core.